### PR TITLE
Add nodeSelector for konnectivity daemonSet

### PR DESCRIPTION
**Pull request by jingxu97**
konnectivity agent daemonSet can only run on Linux node. Add node
selector to the yaml file

**What type of PR is this?**
/kind bug

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by pjh — COMMENTED** [review]
Please clean up the PR description. You can remove the Change-Id. Set the release note properly. Remove any fields that aren't used.

**Comment by yujuhong** [comment]
/approve

@jingxu97 could you cc the maintainers of the konnectivity agent?

**Comment by jingxu97** [comment]
cc @caesarxuchao for this setup on konnectivity

File: test/e2e/storage/drivers/in_tree.go [hunk 1]
```diff
@@ -1269,6 +1268,39 @@ func InitGcePdDriver() testsuites.TestDriver {
 	}
 }
 
+// InitWindowsGcePdDriver returns gcePdDriver running no Windows cluster that implements TestDriver interface
```
Comment (caesarxuchao) [inline test/e2e/storage/drivers/in_tree.go:1271]: nit: s/no/on

**Review by caesarxuchao — COMMENTED** [review]
LGTM. Thanks, @jingxu97

File: test/e2e/storage/drivers/in_tree.go (hunk 1 above)
Comment (jingxu97) [reply test/e2e/storage/drivers/in_tree.go]: sorry, this is a wrong commit, removed it.

**Comment by jingxu97** [comment]
updated the PR. PTAL @caesarxuchao

**Pull request merged**